cohortextractor/main.py

```diff
@@ -1,18 +1,33 @@
 import csv
 import importlib.util
 import inspect
+import shutil
 import sys
 from contextlib import contextmanager
+from pathlib import Path
 
-from .backends import BACKENDS
-from .query_utils import get_column_definitions
+from cohortextractor.backends import BACKENDS
+from cohortextractor.query_utils import get_column_definitions
+from cohortextractor.validate_dummy_data import (
+    DummyDataValidationError,
+    validate_dummy_data,
+)
 
 
-def main(definition_path, output_file, backend_id, db_url):
-    backend = BACKENDS[backend_id](db_url)
+def main(definition_path, output_file, backend_id, db_url, dummy_data_file=None):
     cohort = load_cohort(definition_path)
-    results = extract(cohort, backend)
-    write_output(results, output_file)
+
+    if dummy_data_file:
+        results = None
+        if Path(output_file).suffixes != dummy_data_file.suffixes:
+            expected_extension = "".join(output_file.suffixes)
```

Comment:
It looks odd to have `Path(output_file).suffixes` in the line above and `output_file.suffixes` here. Is it intentional?

Reply:
no, just a copy/paste from the old cohort extractor (which also has this discrepancy, and I assume also isn't intentional). Both `output_file` and `dummy_data_file` should always be a `Path` (assuming they've been called via __main__.py